library/network_connections.py

```diff
@@ -1148,6 +1148,53 @@ def connection_create(self, connections, idx, connection_current=None):
                     s_ip4.add_route(rr)
                 else:
                     s_ip6.add_route(rr)
+            for r in ip["routing_rules"]:
+                routing_rule = NM.IPRoutingRule.new(r["family"])
+                if r["priority"] is not None:
+                    NM.IPRoutingRule.set_priority(routing_rule, r["priority"])
+                if r["table"] is not None:
+                    NM.IPRoutingRule.set_table(routing_rule, r["table"])
+                if r["from"]:
+                    NM.IPRoutingRule.set_from(routing_rule, r["from"], r["from_len"])
+                if r["to"]:
+                    NM.IPRoutingRule.set_to(routing_rule, r["to"], r["to_len"])
+                if r["iif"]:
+                    NM.IPRoutingRule.set_iifname(routing_rule, r["iif"])
+                if r["oif"]:
+                    NM.IPRoutingRule.set_oifname(routing_rule, r["oif"])
+                if r["ipproto"]:
+                    NM.IPRoutingRule.set_ipproto(routing_rule, r["ipproto"])
+                if r["fwmark"]:
+                    NM.IPRoutingRule.set_fwmark(routing_rule, r["fwmark"])
+                if r["sport_start"]:
+                    NM.IPRoutingRule.set_source_port(
+                        routing_rule, r["sport_start"], r["sport_end"]
+                    )
+                if r["dport_start"]:
+                    NM.IPRoutingRule.set_destination_port(
+                        routing_rule, r["dport_start"], r["dport_end"]
+                    )
+                if r["suppress_prefixlength"]:
+                    NM.IPRoutingRule.set_destination_port(
+                        routing_rule, r["suppress_prefixlength"]
+                    )
+                if r["action"]:
+                    for i, action in enumerate(
+                        [
+                            "to-table",
+                            "blackhole",
+                            "prohibit",
+                            "unreachable",
+                        ]
+                    ):
+                        if action == r["action"]:
+                            # check the link below for the enum value of supported action
+                            # https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/uapi/linux/fib_rules.h#n88
+                            NM.IPRoutingRule.set_action(routing_rule, (1, 6, 7, 8)[i])
+                if r["family"] == socket.AF_INET:
+                    s_ip4.add_routing_rule(routing_rule)
+                else:
```

Comment:
The variables here are confusing. AFAICU, 'r' is the routing rule provided by the user and 'routing_rule' is the NM object. Could you improve this? maybe we can have something like 'routing_rule' and 'nm_routing_rule'.

Reply:
This comment is not resolved yet